pkg/sensors/program/loader.go

```diff
@@ -857,6 +857,50 @@ func doLoadProgram(
 	}
 	defer coll.Close()
 
+	// This is for accounting of the BPF map memlock usage. At first I thought I
+	// could just read the coll.Maps but those contain (a lot of) unused maps in
+	// the ELF file (e.g. tg_cgrps_tracking_map) that are loaded initially but
+	// not used or referenced anywhere so GCed as soon as we close the reference
+	// on the collection.
+	//
+	// The way we found is to browse which maps are used from the returned
+	// programs in the collection. Since for those programs we keep a reference
+	// anyway, they can't be garbage collected.
+	collMaps := map[ebpf.MapID]*ebpf.Map{}
+	// we need a mapping by ID
+	for _, m := range coll.Maps {
+		info, err := m.Info()
+		if err != nil {
+			logger.GetLogger().WithError(err).WithField("map", m.String()).Warn("failed to retrieve BPF map info")
+		}
+		id, available := info.ID()
+		if !available {
+			logger.GetLogger().WithField("map", m.String()).Warn("failed to retrieve BPF map ID, you might be running <4.13")
+		}
+		collMaps[id] = m
+	}
+	load.LoadedMapsInfo = map[int]bpf.ExtendedMapInfo{}
+	for _, p := range coll.Programs {
+		i, err := p.Info()
+		if err != nil {
+			logger.GetLogger().WithError(err).WithField("program", p.String()).Warn("failed to retrieve BPF program info, you might be running <4.10")
```

Comment:
should we break here? ditto for the other error checks where we only emit a log message.

Reply:
yes good idea, let's give up there's almost no chance if can only fail one time.